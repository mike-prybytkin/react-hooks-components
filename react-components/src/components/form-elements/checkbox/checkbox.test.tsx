import React from 'react';
import { render, screen } from '@testing-library/react';
import mockText from 'mocks/text';
import Checkbox from './checkbox';

const setUp = () => {
  return render(
    <Checkbox labelType={mockText.labelUserMailing} checkboxRef={undefined} id={'test'} />
  );
};

describe('Checkbox component', () => {
  it('should correctly render checkbox', () => {
    setUp();
    expect(screen.getByRole('checkbox', { checked: true })).toBeInTheDocument();
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
