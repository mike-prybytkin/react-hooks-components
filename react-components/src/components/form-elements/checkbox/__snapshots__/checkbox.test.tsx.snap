// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should create snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkbox"
  >
    <input
      checked=""
      class="checkbox__input"
      id="test"
      type="checkbox"
    />
    <label
      class="checkbox__label"
      for="test"
    >
      Receive mailing
    </label>
  </div>
</DocumentFragment>
`;
